src/silx/conftest.py

```diff
@@ -77,12 +77,18 @@ def pytest_configure(config):
     "ignore:Non-empty compiler output encountered. Set the environment variable PYOPENCL_COMPILER_OUTPUT=1 to see more.:UserWarning",
     # Remove __array__ ignore once h5py v3.12 is released
     "ignore:__array__ implementation doesn't accept a copy keyword, so passing copy=False failed. __array__ must implement 'dtype' and 'copy' keyword arguments.:DeprecationWarning",
-    "ignore::pyopencl.RepeatedKernelRetrieval",
     # Deprecated pyparsing usage in matplotlib: https://github.com/matplotlib/matplotlib/issues/30617
     "ignore::DeprecationWarning:matplotlib._fontconfig_pattern",
     "ignore::DeprecationWarning:matplotlib._mathtext",
     "ignore::DeprecationWarning:pyparsing.util",
 )
+try:
+    import pyopencl
+except:
+    pass
+else:
+    if "RepeatedKernelRetrieval" in dir(pyopencl):
```

Comment:
Why is this needed?

When does this occurs? Is it limited to a few specific tests? if so, best to filter this warning only for those tests with [pytest.mark.filterwarnings](https://docs.pytest.org/en/7.1.x/reference/reference.html?highlight=filterwarnings#pytest.mark.filterwarnings)

Which version of pyopencl have it/does not have it?
If it's is old enough, IMO it reasonable to enforce a minimum version in the dependencies and spare the if statement
If not, it's best to test the version of pyopencl rather than relying on a `hasattr`-like test so it documents why this check is needed for (and when we can get rid of it).

Reply:
Elder version of pyopencl (2025.1) do not have this attribute. Silencing a warning that does not exist simply prevents pytest from launching.